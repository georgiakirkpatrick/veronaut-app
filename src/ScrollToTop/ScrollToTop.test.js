import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter, MemoryRouter } from 'react-router-dom'
import ScrollToTop from './ScrollToTop'

const renderWithRouter = (ui, { route = '/' } = {}) => {
    window.history.pushState({}, 'Test page', route)
    const div = document.createElement('div')

    return ReactDOM.render(
        <BrowserRouter>
            {ui}
        </BrowserRouter>,
        div
    )
}

describe('ScrollToTop', () => {
    it ('renders without crashing', () => {
        const route = '/some-route'
        renderWithRouter(<ScrollToTop />, { route })
    })

    it ('render the UI as expected', () => {
        const tree = renderer
            .create(
                <MemoryRouter>
                    <ScrollToTop />
                </MemoryRouter>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})