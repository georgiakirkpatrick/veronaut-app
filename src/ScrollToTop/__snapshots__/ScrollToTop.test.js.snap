// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTop render the UI as expected 1`] = `null`;
